src/engine/mod.rs

```diff
@@ -22,14 +22,58 @@ pub struct Engine {
 
 impl Default for Engine {
     fn default() -> Engine {
-        let mut filters = HashMap::new();
-        filters.insert("upper".into(), Box::new(filter::upper) as FilterFn);
-        filters.insert("lower".into(), Box::new(filter::lower) as FilterFn);
+        let mut engine = Engine::new();
+        engine.register_builtin_filters();
```

Comment:
Nit - would it make sense to move the `register_builtin...` calls to a separate constructor, like `with_default_builtins` and then call it from here ?

Reply:
More like a thought here - would it be more work if we made the engine immutable and the `register...` methods return a new engine ? I found that it typically leads to clearer user code if things are externally immutable.

Reply:
Nope, there's `Default` and it contains all built ins. `new` is private, user shouldn't create empty engine (without builtins).

Reply:
Re immutability - then I'd introduce something like `EngineBuilder`. I'll do it to show to you what I mean. Give me few minutes.